File: clients/PHP-client/src/TTLClient.php
<?php

namespace SoliDB;

class TTLClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function createIndex(string $collection, string $name, string $field, int $expireAfterSeconds): array
    {
        $res = $this->client->sendCommand('create_ttl_index', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'name' => $name,
            'field' => $field,
            'expire_after_seconds' => $expireAfterSeconds
        ]);
        return $res['data'] ?? [];
    }

    public function listIndexes(string $collection): array
    {
        $res = $this->client->sendCommand('list_ttl_indexes', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ]);
        return $res['data'] ?? [];
    }

    public function deleteIndex(string $collection, string $indexName): void
    {
        $this->client->sendCommand('delete_ttl_index', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'index_name' => $indexName
        ]);
    }

    public function updateExpiration(string $collection, string $indexName, int $expireAfterSeconds): void
    {
        $this->client->sendCommand('update_ttl_expiration', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'index_name' => $indexName,
            'expire_after_seconds' => $expireAfterSeconds
        ]);
    }

    public function getIndexInfo(string $collection, string $indexName): array
    {
        $res = $this->client->sendCommand('ttl_index_info', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection,
            'index_name' => $indexName
        ]);
        return $res['data'] ?? [];
    }

    public function runCleanup(string $collection): array
    {
        $res = $this->client->sendCommand('ttl_run_cleanup', [
            'database' => $this->client->getDatabase(),
            'collection' => $collection
        ]);
        return $res['data'] ?? [];
    }
}

File: clients/Ruby-client/lib/solidb/ttl_client.rb
module SoliDB
  class TTLClient
    def initialize(client)
      @client = client
    end

    def create_index(collection, name:, field:, expire_after_seconds:)
      @client.send_command("create_ttl_index",
        database: @client.database,
        collection: collection,
        name: name,
        field: field,
        expire_after_seconds: expire_after_seconds
      )
    end

    def list_indexes(collection)
      @client.send_command("list_ttl_indexes", database: @client.database, collection: collection) || []
    end

    def delete_index(collection, index_name)
      @client.send_command("delete_ttl_index", database: @client.database, collection: collection, index_name: index_name)
      nil
    end

    def update_expiration(collection, index_name, expire_after_seconds)
      @client.send_command("update_ttl_expiration",
        database: @client.database,
        collection: collection,
        index_name: index_name,
        expire_after_seconds: expire_after_seconds
      )
      nil
    end

    def get_index_info(collection, index_name)
      @client.send_command("ttl_index_info", database: @client.database, collection: collection, index_name: index_name)
    end

    def run_cleanup(collection)
      @client.send_command("ttl_run_cleanup", database: @client.database, collection: collection)
    end
  end
end
